pkg/kubectl/cmd/exec/exec.go

```diff
@@ -266,11 +288,35 @@ func (o *StreamOptions) SetupTTY() term.TTY {
 
 // Run executes a validated remote execution against a pod.
 func (p *ExecOptions) Run() error {
-	pod, err := p.PodClient.Pods(p.Namespace).Get(p.PodName, metav1.GetOptions{})
-	if err != nil {
-		return err
+	var err error
+	// we still need legacy pod getter when PodName in ExecOptions struct is provided,
+	// since there are any other command run this function by providing Podname with PodsGetter
+	// and without resource builder, eg: `kubectl cp`.
+	if len(p.PodName) != 0 {
+		p.Pod, err = p.PodClient.Pods(p.Namespace).Get(p.PodName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+	} else {
+		builder := p.Builder().
+			WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+			NamespaceParam(p.Namespace).DefaultNamespace()
+
+		builder.ResourceNames("pods", p.ResourceName)
```

Comment:
nit: chain this with the rest of the builder methods you execute above

Reply:
@kargakis i will update it soon

Reply:
updated
thanks @kargakis